fb_apt/providers/keys.rb

```diff
@@ -0,0 +1,69 @@
+# vim: syntax=ruby:expandtab:shiftwidth=2:softtabstop=2:tabstop=2
+#
+# Copyright (c) 2016-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+#
+
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+action :run do
+  keyring = node['fb_apt']['keyring']
+  keyserver = node['fb_apt']['keyserver']
+  keys = node['fb_apt']['keys'].to_hash
+
+  if keys && keyring
+    if ::File.exists?(keyring)
+      cmd = Mixlib::ShellOut.new("LANG=C apt-key --keyring #{keyring} list")
+      cmd.run_command
+      cmd.error!
+      output = cmd.stdout.split("\n")
+      Chef::Log.debug("apt-key output: #{output.join("\n")}")
+      installed_keys = output.select { |x| x.start_with?('pub') }.map do |x|
+        x[%r/pub.*\/(?<keyid>[A-Z0-9]*)/, 'keyid']
+      end
+      Chef::Log.info("Installed keys: #{installed_keys.join(', ')}")
```

Comment:
I'm fine with this code, but it came be made a bit simpler with:
```
installed_keys = []
if File.exists?(keyring)
  ...
  installed_keys = ...
end
Chef::Log.debug("Installed keys: ...")
```

No need for different print statements, you have a default, etc.

Also, that should be debug.